pkg/operator/utils/config.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	opv1 "github.com/openshift/api/operator/v1"
+	"strconv"
+)
+
+func GetSnapshotOptions(clusterCSIDriver *opv1.ClusterCSIDriver) map[string]string {
+	snapshotOptions := map[string]string{}
+	if clusterCSIDriver == nil || clusterCSIDriver.Spec.DriverConfig.VSphere == nil {
+		return snapshotOptions
+	}
+
+	vSphereConfig := clusterCSIDriver.Spec.DriverConfig.VSphere
+
+	options := []string{
+		"global-max-snapshots-per-block-volume",
+		"granular-max-snapshots-per-block-volume-vsan",
+		"granular-max-snapshots-per-block-volume-vvol",
+	}
+
+	for _, option := range options {
+		var value *uint32
+
+		switch option {
+		case "global-max-snapshots-per-block-volume":
+			value = vSphereConfig.GlobalMaxSnapshotsPerBlockVolume
+		case "granular-max-snapshots-per-block-volume-vsan":
+			value = vSphereConfig.GranularMaxSnapshotsPerBlockVolumeInVSAN
+		case "granular-max-snapshots-per-block-volume-vvol":
+			value = vSphereConfig.GranularMaxSnapshotsPerBlockVolumeInVVOL
+		}
+
+		if value != nil {
+			snapshotOptions[option] = strconv.FormatUint(uint64(*value), 10)
+		}
+	}
```

Comment:
Why is it so complicated? Why do you iterate over an array of options to have a `case` for each of the option?!

```go
func addOption(options map[string]string, name string, intValue *uint32) {
    if stringValue == nil { return }
    options[name] = strconv.FormatUint(uint64(*value), 10)
}

func GetSnapshotOptions(clusterCSIDriver *opv1.ClusterCSIDriver) map[string]string {
    snapshotOptions := map[string]string{}
    if clusterCSIDriver == nil || clusterCSIDriver.Spec.DriverConfig.VSphere == nil { return }

    addOption(snapshotOptions, "global-max-snapshots-per-block-volume", vSphereConfig.GlobalMaxSnapshotsPerBlockVolume)
   ...
}

Reply:
Seemed slightly better than calling the function repeatedly, but this is more readable - changed.